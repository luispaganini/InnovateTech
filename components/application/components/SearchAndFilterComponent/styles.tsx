import { ThemedView } from "@/components/ThemedView";
import styled from "styled-components/native";

export const SearchAndFilterContainer = styled(ThemedView)`
    flex-direction: row;
    align-items: center;
    display: flex;
    justify-content: space-between;
    width: 90%;
    align-self: center;
    margin-bottom: 10px;
    margin-top: 10px;
`;

export const FilterIconContainer = styled(ThemedView)`
    color: ${props => props.theme.text};
    padding-left: 10px;
`;