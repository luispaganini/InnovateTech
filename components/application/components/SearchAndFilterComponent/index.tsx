import React from 'react'
import SearchBarComponent from '../SearchBarComponent'
import { useColorScheme } from '@/hooks/useColorScheme';
import { Colors } from '@/constants/Colors';
import { MaterialIcons } from '@expo/vector-icons';
import { FilterIconContainer, SearchAndFilterContainer } from './styles';

interface SearchAndFilterComponentProps {
    setQuery: (query: string) => void;
    query: string;
    onSubmit: () => void;
    onClickFilter: () => void;
}

export default function SearchAndFilterComponent(props: SearchAndFilterComponentProps) {
    const colorScheme = useColorScheme();
    return (
        <SearchAndFilterContainer>
            <SearchBarComponent setQuery={props.setQuery} query={props.query} onSubmit={props.onSubmit} />
            <FilterIconContainer>
                <MaterialIcons name="filter-list-alt" size={30} color={colorScheme == 'light' ? Colors.light.text : Colors.dark.text} onPress={props.onClickFilter} />
            </FilterIconContainer>
        </SearchAndFilterContainer>
    )
}