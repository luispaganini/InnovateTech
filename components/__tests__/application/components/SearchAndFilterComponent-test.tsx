import SearchAndFilterComponent from '@/components/application/components/SearchAndFilterComponent';
import React from 'react';
import renderer from 'react-test-renderer';

describe('SearchAndFilterComponent', () => {
  it('renders correctly', () => {
    const setQueryMock = jest.fn();
    const onClickFilterMock = jest.fn();
    const tree = renderer
      .create(
        <SearchAndFilterComponent
          setQuery={setQueryMock}
          query=""
          onClickFilter={onClickFilterMock}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls setQuery when SearchBarComponent changes query', () => {
    const setQueryMock = jest.fn();
    const onClickFilterMock = jest.fn();
    const component = renderer.create(
      <SearchAndFilterComponent
        setQuery={setQueryMock}
        query=""
        onClickFilter={onClickFilterMock}
      />
    );
    const searchBar = component.root.findByProps({ testID: 'search-bar' });

    searchBar.props.setQuery('new query');

    expect(setQueryMock).toHaveBeenCalledWith('new query');
  });

  it('calls onClickFilter when FilterIconContainer is pressed', () => {
    const setQueryMock = jest.fn();
    const onClickFilterMock = jest.fn();
    const component = renderer.create(
      <SearchAndFilterComponent
        setQuery={setQueryMock}
        query=""
        onClickFilter={onClickFilterMock}
      />
    );
    const filterIcon = component.root.findByProps({ testID: 'filter-icon' });

    filterIcon.props.onPress();

    expect(onClickFilterMock).toHaveBeenCalledTimes(1);
  });
});
