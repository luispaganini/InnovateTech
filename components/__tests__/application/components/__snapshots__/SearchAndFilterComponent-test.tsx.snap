// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAndFilterComponent renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "center",
        "alignSelf": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 10,
        "marginTop": 10,
        "width": "90%",
      },
    ]
  }
>
  <View
    colorTheme="light"
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "borderBottomLeftRadius": 2,
          "borderBottomRightRadius": 2,
          "borderColor": "#687076",
          "borderStyle": "solid",
          "borderTopLeftRadius": 2,
          "borderTopRightRadius": 2,
          "borderWidth": 1,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 5,
          "paddingLeft": 10,
          "paddingRight": 5,
          "paddingTop": 5,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <TextInput
        colorTheme="light"
        onChangeText={[Function]}
        placeholder="Busca..."
        placeholderTextColor="#11181C"
        style={
          {
            "backgroundColor": "#fff",
            "color": "#11181C",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 2,
            "paddingLeft": 2,
            "paddingRight": 2,
            "paddingTop": 2,
          }
        }
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "color": "",
          "paddingLeft": 10,
        },
      ]
    }
  >
    <Text />
  </View>
</View>
`;
